pkg/scheduler/reconciliation/inmemory.go

```diff
@@ -240,3 +242,31 @@ func (r *InMemoryReconciliationRepository) UpdateOperationState(schedulingID, co
 
 	return nil
 }
+
+func (r *InMemoryReconciliationRepository) UpdateOperationRetryID(schedulingID, correlationID, retryID string) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	_, ok := r.operations[schedulingID]
+	if !ok {
+		return &repository.EntityNotFoundError{}
+	}
+	op, ok := r.operations[schedulingID][correlationID]
+	if !ok {
+		return &repository.EntityNotFoundError{}
+	}
+
+	// copy the operation to avoid having data races while writing
+	opCopy := *op
```

Comment:
Not needed.. there cannot be data-races as we use a mutex above

Reply:
I copied the code from the `UpdateOperationState` method implementation, which is also using a lock. should we skip the copy step in the `UpdateOperationState` method too?

Reply:
ok, that would be good. Please drop this part form the other fct as well. I don't think that a copy will make anything better. A copy won't avoid race-conditions when two routines are changing the same operations... finally, the last one will overwrite with his copy the change of the other one